### PR TITLE
feat: navbar icon

### PR DESCRIPTION
**Pull request by benmelz**
Built off #2

Adds a little fontawesome icon to the navbar. I'm thinking we might obviously come up with a fully custom icon eventually, but I figured we could use a placeholder for a _little_ character.

Also, fontawesome was broken (the initializer/`fa-font-path` adjustments fixes it). Forgot what version of rails this was.

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -12,3 +12,5 @@
 # application.js, application.css, and all non-JS/CSS in the app/assets
 # folder are already added.
 # Rails.application.config.assets.precompile += %w( admin.js admin.css )
+
+Rails.application.config.assets.paths << Rails.root.join('node_modules/@fortawesome/fontawesome-free/webfonts')
```
Comment (hugoamaralsantana) [inline config/initializers/assets.rb:16]: I'm guessing this just latches on some JS fontAwesome things to the Rails assets?

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -11,6 +11,8 @@
     %header
       %nav.navbar.navbar-dark.bg-dark.border-bottom
         .container-fluid
-          = link_to 'WAV3', root_path, class: 'navbar-brand px-3'
+          = link_to root_path, class: 'navbar-brand px-2' do
```
[the remark this answers is not in the document]
Comment (benmelz) [reply app/views/layouts/application.html.haml:14]: Yeah, a lot of rails HTML helpers (i.e. methods that generate HTML tags) allow you to pass blocks like this (block being the `do` portion) to specify what should be nested in it

File: config/initializers/assets.rb (hunk 1 above)
Comment (benmelz) [reply config/initializers/assets.rb:16]: Yeah it's about "resolving paths" and such. The fontawesome source code has some definitions like

```
@font-face {
  ...
  src: url('fa-regular-400.woff2') ... ,
  ...
}
```

so the asset compiler is going to try to look for a file named `fa-regular-400.woff2`.

This line adds the correct directory to the list paths the compiler is going to look for that file in.

File: app/views/layouts/application.html.haml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hugoamaralsantana) [reply app/views/layouts/application.html.haml:14]: Ah okay, definitely going to take a good week or two of constantly referencing the docs for every little thing but I'll get the hang of it quick I think. The work I did with Django helps a lot

File: config/initializers/assets.rb (hunk 1 above)
Comment (hugoamaralsantana) [reply config/initializers/assets.rb:16]: I sort of understood it but at the same time I didn't completely understand why it couldn't be added in the .scss file. I'll chalk it up to config stuff that'll go over my head in the meantime that isn't important enough for me to worry about now, but I'll pick it up in the future

**Review by hugoamaralsantana — APPROVED** [review]
Learning bit by bit 💯

File: config/initializers/assets.rb (hunk 1 above)
Comment (benmelz) [reply config/initializers/assets.rb:16]: Don't sweat it - the asset pipeline (js + css + images + fonts) is IMO one of the hardest things to wrap your head around with rails. You also don't have to deal with it _too_ often (usually it just works) unless you're working with libraries like this. It took me like a year of fiddling to really get down how it all works.

**Pull request merged**